recipe/meta.yaml

```
@@ -20,9 +20,10 @@ build:
 requirements:
   host:
     - pip
-    - python >=3.6
+    - python >=3.7
   run:
-    - py-opencv <=4.9
+    - numpy <2.0.0
```

Could you elaborate on the `numpy<2` upper pin here? I saw https://github.com/obss/sahi/pull/1057, but it doesn't mention any errors or what is incompatible.

The issue is that users can still conda install `numpy=2` with `sahi=0.11.16` or older, since the previous versions did not set this pin. There's a way to prevent that by submitting a repodata-patch to https://github.com/conda-forge/conda-forge-repodata-patches-feedstock, but want to get more context before doing all that work :slightly_smiling_face:

---

There are breaking changes in the recently released numpy version that collides with precompiled pytorch binaries. That's why I had to add that constrain @weiji14

---

Please check this action run for the error I mention: https://github.com/obss/sahi/actions/runs/9866647136/job/27245723778

---

Hmm, I don't think it's necessarily an issue on `sahi`, but rather on opencv. The error seems to be on the `import cv2` line:

```python-traceback
Traceback (most recent call last):  File "/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
...
  File "/home/runner/work/sahi/sahi/sahi/__init__.py", line 3, in <module>
    from sahi.annotation import BoundingBox, Category, Mask
  File "/home/runner/work/sahi/sahi/sahi/annotation.py", line 10, in <module>
    from sahi.utils.cv import (
  File "/home/runner/work/sahi/sahi/sahi/utils/cv.py", line 10, in <module>
    import cv2
AttributeError: _ARRAY_API not found
E..
A module that was compiled using NumPy 1.x cannot be run in
NumPy 2.0.0 as it may crash. To support both 1.x and 2.x
versions of NumPy, modules must be compiled with NumPy 2.0.
Some module may need to rebuild instead e.g. with 'pybind11>=2.12'.

If you are a user of the module, the easiest solution will be to
downgrade to 'numpy<2' or try to upgrade the affected module.
We expect that some modules will need time to support NumPy 2.
```

So we need a version of opencv compiled with NumPy 2.0, but that is only available with OpenCV 4.10.0 (https://github.com/opencv/opencv-python/issues/943, https://github.com/conda-forge/opencv-feedstock/pull/417), and sahi uses `opencv<=4.9`, hence why numpy 2.0 doesn't work.

In this case, the repodata-patch should probably be applied on opencv instead of sahi. I.e., we could remove the `numpy<2.0` pin here, and have the `numpy<2.0` be set in https://github.com/conda-forge/opencv-feedstock instead. Give me a few hours to open PRs for that.

---

Hmm actually, [`py-opencv=4.9.0`](https://anaconda.org/conda-forge/py-opencv/files?version=4.9.0) already has a version constraint for `numpy<2.0a0` (set in https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/728, so we could probably just remove the `numpy<2.0` pin here and things should work (specifically, `sahi` on conda-forge won't be able to install numpy=2.0 because `py-opencv` can't):

```suggestion
    - numpy
```

---

Yep, CI at https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=976316&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=986b1512-c876-5f92-0d81-ba851554a0a3&l=1159 installs `numpy=1.26.4` as expected instead of `numpy=2.0`, should be all good now!